Desktop/engine/enginerepresentation.cpp

```diff
@@ -1119,6 +1119,9 @@ void EngineRepresentation::sendReloadData()
 
 void EngineRepresentation::addSettingsToJson(Json::Value & msg)
 {
+	if(!PreferencesModel::prefs()) //During testing only!
```

Comment:
Why did you not create a new PreferencesModelBase() in the testengine initialization? You would not have to add all these if's